.github/tsc.json

```diff
@@ -4,14 +4,15 @@
             "owner": "tsc",
             "pattern": [
                 {
-                    "regexp": "^(?:\\s+\\d+\\>)?([^\\s].*)\\((\\d+|\\d+,\\d+|\\d+,\\d+,\\d+,\\d+)\\)\\s*:\\s+(error|warning|info)\\s+(\\w{1,2}\\d+)\\s*:\\s*(.*)$",
+                    "regexp": "^([^\\s].*)[\\(:](\\d+)[,:](\\d+)(?:\\):\\s+|\\s+-\\s+)(error|warning|info)\\s+TS(\\d+)\\s*:\\s*(.*)$",
```

Comment:
It might be good to add a unit test that loads the regexp and processes it against a line, asserts expected match groups

I'm worried if this change regresses something, at least we have a baseline unit test suite going forward.

Also i'm curious how this compares to vscode's built-in `tsc` problem matcher or their `tsc-watch` problem matcher. I briefly looked, but wan't able to find those built-in regexps in their repo.

Reply:
@ericsciple looks like tests are below.  have a quick look